api/types/duration.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/teleport"
+
+	"github.com/gravitational/trace"
+)
+
+// Duration is a wrapper around duration to set up custom marshal/unmarshal
+type Duration time.Duration
+
+// Duration returns time.Duration from Duration typex
+func (d Duration) Duration() time.Duration {
+	return time.Duration(d)
+}
+
+// Value returns time.Duration value of this wrapper
+func (d Duration) Value() time.Duration {
+	return time.Duration(d)
+}
+
+// MarshalJSON marshals Duration to string
+func (d Duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(fmt.Sprintf("%v", d.Duration()))
+}
+
+// UnmarshalJSON marshals Duration to string
```

Comment:
```suggestion
// UnmarshalJSON marshals this duration value as a string
```
for naming consistency